daemon/daemon.go

```diff
@@ -1520,6 +1515,33 @@ func (daemon *Daemon) Reload(config *Config) error {
 	if config.IsValueSet("debug") {
 		daemon.configStore.Debug = config.Debug
 	}
+
+	// If no value is set for max-downloads we assume it is the default value
+	// We always "reset" as the cost is lightweight and easy to maintain.
+	if config.IsValueSet("max-downloads") && config.MaxDownloads != nil {
+		daemon.configStore.MaxDownloads = config.MaxDownloads
```

Comment:
This is probably fine in practice, but I think it would be slightly better to do:

```go
*daemon.configStore.MaxDownloads = *config.MaxDownloads
```

so that if something changes `config.MaxDownloads` after calling `Reload`, it doesn't unexpectedly change `daemon.configStore.MaxDownloads`.